mongo/index_view.go

```diff
@@ -45,8 +44,9 @@ type IndexView struct {
 
 // IndexModel represents a new index to be created.
 type IndexModel struct {
-	// A document describing which keys should be used for the index. It cannot be nil. See
-	// https://docs.mongodb.com/manual/indexes/#indexes for examples of valid documents.
+	// A document describing which keys should be used for the index. It cannot be nil. This must be an order-preserving
+	// type such as bson.D. Map types such as bson.M are not valid. See https://docs.mongodb.com/manual/indexes/#indexes
```

Comment:
Is this a new restriction, or documentation of an existing restriction?

Reply:
Using maps for the key spec has always been incorrect because Go maps aren't ordered, so this is documentation of an existing restriction.